lib/Syntax/SyntaxFactory.cpp.gyb

```diff
@@ -42,15 +42,15 @@ TokenSyntax SyntaxFactory::makeToken(tok Kind, OwnedString Text,
                                      const Trivia &LeadingTrivia,
                                      const Trivia &TrailingTrivia,
                                      SourcePresence Presence,
-                                     SyntaxArena *Arena) {
+                                     RC<SyntaxArena> Arena) {
```

Comment:
Good idea. I'll change it where possible.

Reply:
I've changed it in many places, however, if the argument has type `RC<SyntaxArena> &` it can no longer have `nullptr` as its default value. 

I decided to create methods without an `Arena` argument for `RawSyntax::make` that just dispatch to the `make` function with the `Arena` argument, passing a `nullptr` for the arena (see [RawSyntax.h](https://github.com/apple/swift/pull/18886/files#diff-098952e63a5993652ba35305d0120679R355)), but decided that the `SyntaxFactory` methods are not performance-critical enough to create a second copy of every `make*` function.